core/templates/services/literally-canvas-helper.service.ts

```diff
@@ -58,31 +58,240 @@ angular.module('oppia').factory('LiterallyCanvasHelperService', [
+          if (typeof strokeDash !== 'undefined') {
+            shape.data.dash = strokeDash.value.split(', ').map(
+              a => parseInt(a));
+          }
+          shape.data.isClosed = false;
+          var shift = strokeWidth % 2 !== 0 ? 0.5 : 0;
+          shape.data.pointCoordinatePairs = (
+            innerPolygon.attributes.points.value.split(' ').map(
+              a => a.split(',').map(b => parseFloat(b) - shift)));
+        }
+        shape.id = id.slice(2).join('-');
+      } else if (id[0] === 'text') {
+        shape.className = 'Text';
+        shape.data.x = parseFloat(node.attributes.x.value);
+        shape.data.y = parseFloat(node.attributes.y.value);
+        var text = '';
+        node.querySelectorAll('*').forEach(a=>{
+          text += a.textContent + '\n';
+        });
+        shape.data.text = text.slice(0, -1);
+        shape.data.color = node.attributes.fill.value;
+        shape.data.font = node.attributes.style.value.slice(6, -1);
+        shape.data.forcedWidth = (
+          typeof node.attributes.width !== 'undefined' ? (
+            parseFloat(node.attributes.width.value)) : 0);
+        shape.data.forcedHeight = (
+          typeof node.attributes.height !== 'undefined' ? (
+            parseFloat(node.attributes.height.value)) : 0);
+        shape.id = id.slice(1).join('-');
+      }
+      return shape;
+    };
+
     return {
-      isSVGTagValid: function(svgString) {
+      parseSvg: function(svgString, lc) {
+        // This function is used to convert an svg to a snapshot object which
+        // is used by literallyCanvas to load the svg into canvas.
         var domParser = new DOMParser();
-        var doc = domParser.parseFromString(svgString, 'image/svg+xml');
-        var allowedTags = Object.keys(CONSTANTS.SVG_ATTRS_WHITELIST);
-        var nodeTagName = null;
-        var valid = true;
-        doc.querySelectorAll('*').forEach((node) => {
-          nodeTagName = node.tagName.toLowerCase();
-          if (allowedTags.indexOf(nodeTagName) !== -1) {
-            for (var i = 0; i < node.attributes.length; i++) {
-              if (CONSTANTS.SVG_ATTRS_WHITELIST[nodeTagName].indexOf(
-                node.attributes[i].name.toLowerCase()) === -1) {
-                valid = false;
-              }
-            }
-          } else {
-            valid = false;
+        var doc = domParser.parseFromString(svgString, 'text/xml');
+        var snapshot = {
+          colors: {
+            primary: '',
+            secondary: '',
+            background: ''
+          },
+          position: '',
+          scale: 1,
+          shapes: [],
+          backgroundShapes: [],
+          imageSize: {
+            width: '',
+            height: ''
           }
+        };
+        var rect = doc.querySelector('svg > rect');
+        snapshot.colors.primary = lc.colors.primary;
+        snapshot.colors.secondary = lc.colors.secondary;
+        // @ts-ignore fill property doesnot exist error.
```

Comment:
doesnot --> does not, here and in multiple places below.

Also this needs more justification. Why is the ts-ignore necessary? /cc @ankita240796 @nishantwrp

Reply:
Changed from doesnot to does not.

Reply:
So, the problem here is that `attributes` has type `NameNodeMap` and this property doesn't have any properties (Reference - https://developer.mozilla.org/en-US/docs/Web/API/NamedNodeMap). So, maybe instead of this a custom type for `NameNodeMap` should be declared in the `typings` directory which would have the properties which are used here. Similar to what we do for `ng.IScope` in some controllers. For example
https://github.com/oppia/oppia/blob/d12d515d0b65c0aaf126fd17e2ef62fbf44c158d/core/templates/components/ck-editor-helpers/ck-editor-4-rte.directive.ts#L30-L32